contracts/CourtPresaleActivate.sol

```diff
@@ -51,8 +57,72 @@ contract CourtPresaleActivate is IsContract, ApproveAndCallFallBack {
         // move tokens to this contract
         require(ERC20(_token).safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
 
+        _buyAndActivate(_from, _amount, _token);
+    }
+
+    /**
+    * @dev This function needs a previous approval on the external token used for the contributed amount.
+    *      It will pull the approved tokens, convert them in Uniswap to the presale contribution token,
+    *      convert the obtained tokens in the presale instance, and activate the converted tokens in the
+    *      jurors registry instance of the Aragon Court.
+    * @param _token Address of the external contribution token used
+    * @param _from Address of the original caller (juror) converting and activating the tokens
+    * @param _amount Amount of contribution tokens to be converted and activated
+    * @param _minTokens Minimum amount of presale contribution tokens obtained in Uniswap
+    * @param _minEth Minimum amount of ETH obtained in Uniswap (Uniswap internally converts first to ETH and then to target token)
+    * @param _deadline Transaction deadline for Uniswap
+    */
+    function contributeExternalToken(
+        address _token,
+        address _from,
+        uint256 _amount,
+        uint256 _minTokens,
+        uint256 _minEth,
+        uint256 _deadline
+    )
+        external
+    {
+        require(_amount > 0, ERROR_ZERO_AMOUNT);
+
+        // move tokens to this contract
+        require(ERC20(_token).safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
+
+        // get the Uniswap exchange for the contribution token
+        address contributionTokenAddress = address(presale.contributionToken());
+        IUniswapExchange uniswapExchange = IUniswapExchange(uniswapFactory.getExchange(contributionTokenAddress));
```

Comment:
🚨 From reading the Uniswap source, it is actually the other way around. We should be calling the exchange contract for `_token` and passing `presale.contributionToken()` as the output token:

```suggestion
        IUniswapExchange uniswapExchange = IUniswapExchange(uniswapFactory.getExchange(_token));
```